### PR TITLE
Added selectAll markdown file

### PR DESCRIPTION
**Pull request by craglin**
### Description

I made a term entry for the d3 docs selectAll() function as part of my Git/Github learning module with Codecademy. Closes issue #3955.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@craglin I left some changes for you to make. Try to follow the guidelines for making a contribution. You can check out the [documentation ](https://github.com/Codecademy/docs/tree/main/documentation)for that. Ping me when you are done!!

File: content/d3/concepts/selection/terms/selectAll/selectAll.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+CatalogContent:
+---
+
+**selectAll()** is a function within the d3 library that selects all elements matching the provided parameter. The parameter can be a string to match, or a pre-existing reference to a node. selectAll() will match against all elements that meet the criteria, as opposed to select(), which only matches the first element.
+
+## Syntax
+
+```pseudo
+d3.selectAll([string]|[node])
+```
```
[the remark this answers is not in the document]
Comment (craglin) [reply content/d3/concepts/selection/terms/selectAll/selectAll.md]: Added the description bullet points.

File: content/d3/concepts/selection/terms/selectAll/selectAll.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+## Examples
+
+```js
+const p = d3.selectAll("p")
+```
```
[the remark this answers is not in the document]
Comment (craglin) [reply content/d3/concepts/selection/terms/selectAll/selectAll.md:31]: Added a line above the example describing it.

File: content/d3/concepts/selection/terms/selectAll/selectAll.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+```js
+d3.select(document.body).style("background", "red");
+```
+
+## Additional Resources
+[d3 documentation](https://d3js.org/d3-selection/selecting#selectAll "Official documentation")
+
+[d3 code](https://github.com/d3/d3-selection/blob/main/src/selectAll.js "GitHub Repository")
```
[the remark this answers is not in the document]
Comment (craglin) [reply content/d3/concepts/selection/terms/selectAll/selectAll.md]: Removed additional resource section.

File: content/d3/concepts/selection/terms/selectAll/selectAll.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+Description: A brief definition and examples of using selectAll()
+Subjects: 
+  - 'Data Visualization'
+  - 'Web Design'
+Tags:
+  - 'D3'
+CatalogContent:
+---
+
+**selectAll()** is a function within the d3 library that selects all elements matching the provided parameter. The parameter can be a string to match, or a pre-existing reference to a node. selectAll() will match against all elements that meet the criteria, as opposed to select(), which only matches the first element.
```
[the remark this answers is not in the document]
Comment (craglin) [reply content/d3/concepts/selection/terms/selectAll/selectAll.md]: Updated introduction to use backticks for code functions.

**Comment by craglin** [comment]
@avdhoottt I've made your requested changes, but I don't know about the slugs for the catalogue content. I'm not sure what each slug represents, so I don't know how to specifically put the slugs in for D3. Any suggestions you have would be very much appreciated.

File: content/d3/concepts/selection/terms/selectAll/selectAll.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+Title: 'selectAll()'
+Description: 'A brief definition and examples of using selectAll()'
+Subjects: 
+  - 'Data Visualization'
+  - 'Web Design'
+Tags:
+  - 'D3'
+CatalogContent:
+  - 'introduction-to-javascript'
+  - 'paths/front-end-engineer-career-path'
```
Comment (avdhoottt) [inline content/d3/concepts/selection/terms/selectAll/selectAll.md:11]:
```
- 'learn-d3'
- 'paths/data-science'
```
Add these slugs for the CatalogContent.

File: content/d3/concepts/selection/terms/selectAll/selectAll.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
+---
+Title: 'selectAll()'
+Description: 'A brief definition and examples of using selectAll()'
```
Comment (avdhoottt) [inline content/d3/concepts/selection/terms/selectAll/selectAll.md:3]:
```suggestion
Title: '.selectAll()'
Description: 'A brief definition and examples of using selectAll()'
```
Change the description and add the actual description of the method in one line.

File: content/d3/concepts/selection/terms/selectAll/selectAll.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+`selectAll()`is a function within the d3 library that selects all elements matching the provided parameter. selectAll() will match against all elements that meet the parameter, as opposed to `select()`, which only matches the first element.
+
+## Syntax
+
+```pseudo
+d3.selectAll(item)
+```
+
++ The item can be a string version of an element
++ The item can also be a saved node or array of nodes
```
Comment (avdhoottt) [inline content/d3/concepts/selection/terms/selectAll/selectAll.md:23]: Add the parameters in bullet points, here's an example - 
`- url: the URL of the CSV file or a string containing the CSV data.`

File: content/d3/concepts/selection/terms/selectAll/selectAll.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+## Examples
```
Comment (avdhoottt) [inline content/d3/concepts/selection/terms/selectAll/selectAll.md:25]:
```suggestion
## Example
```

File: content/d3/concepts/selection/terms/selectAll/selectAll.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+This code selects all paragraph elements within the document.
```
Comment (avdhoottt) [inline content/d3/concepts/selection/terms/selectAll/selectAll.md:27]:
```suggestion
The following code selects all paragraph elements within the document using the `.selectAll()` method.
```

File: content/d3/concepts/selection/terms/selectAll/selectAll.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+```js
+const p = d3.selectAll("p")
+```
+
+This code selects the entire document body and applies the style change.
+
+```js
+d3.select(document.body).style("background", "red");
+```
```
Comment (avdhoottt) [inline content/d3/concepts/selection/terms/selectAll/selectAll.md:37]: This is an example of `.select()`, you can remove it as you've already added an example. We need an example of `.selectAll()` method only.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@craglin Some more changes from my side.

**Comment by avdhoottt** [comment]
@craglin It's been a week since I have suggested the changes. Are you working on this PR or not?

File: content/d3/concepts/selection/terms/selectAll/selectAll.md (hunk 7 above)
Comment (craglin) [reply content/d3/concepts/selection/terms/selectAll/selectAll.md:23]: Modified parameter to bullet point, as requested.

File: content/d3/concepts/selection/terms/selectAll/selectAll.md (hunk 10 above)
Comment (craglin) [reply content/d3/concepts/selection/terms/selectAll/selectAll.md:37]: Removed.

**Review by craglin — COMMENTED** [review]
Made changes to document to reflect additional CatalogContent and removed reference to `.select()` method.

**Comment by craglin** [comment]
> @craglin It's been a week since I have suggested the changes. Are you working on this PR or not?

Sorry about that. I had to go to the ER for hand surgery and have had to avoid using my left hand for a while. I'm only recently recovered enough to catch up on work, let alone contributions here.

File: content/d3/concepts/selection/terms/selectAll/selectAll.md [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+Subjects: 
+  - 'Data Visualization'
+  - 'Web Design'
+Tags:
+  - 'D3'
+CatalogContent:
+  - 'introduction-to-javascript'
+  - 'paths/front-end-engineer-career-path'
+  - 'learn-d3' 
+  - 'paths/data-science'
```
Comment (avdhoottt) [inline content/d3/concepts/selection/terms/selectAll/selectAll.md:13]:
```suggestion
  - 'learn-d3' 
  - 'paths/data-science'
```

File: content/d3/concepts/selection/terms/selectAll/selectAll.md [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+  - 'Data Visualization'
+  - 'Web Design'
+Tags:
+  - 'D3'
+CatalogContent:
+  - 'learn-d3' 
+  - 'paths/data-science'
+---
+
+`selectAll()`is a function within the d3 library that selects all elements matching the provided parameter. selectAll() will match against all elements that meet the parameter, as opposed to `select()`, which only matches the first element.
```
Comment (avdhoottt) [inline content/d3/concepts/selection/terms/selectAll/selectAll.md:14]:
```suggestion
The **`.selectAll()`** is a function within the d3 library that selects all elements matching the provided parameter. selectAll() will match against all elements that meet the parameter, as opposed to `.select()`, which only matches the first element.
```

File: content/d3/concepts/selection/terms/selectAll/selectAll.md [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+
+## Syntax
+
+```pseudo
+d3.selectAll(selector)
+```
+
+- selector: a string version of an element, a saved node, or an array of nodes
```
Comment (avdhoottt) [inline content/d3/concepts/selection/terms/selectAll/selectAll.md:22]:
```suggestion
- `selector`: Its a string version of an element, a saved node, or an array of nodes.
```

File: content/d3/concepts/selection/terms/selectAll/selectAll.md [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+
+## Example
+
+The following code selects all paragraph elements within the document using the `.selectAll()` method.
+
+```js
+const p = d3.selectAll("p")
+```
```
Comment (avdhoottt) [inline content/d3/concepts/selection/terms/selectAll/selectAll.md:30]: Add an example such that it'll give an output. Add that output in the ` ```shell ` code block after the code.

File: content/d3/concepts/selection/terms/selectAll/selectAll.md [hunk 15]
```diff
@@ -0,0 +1,30 @@
+---
+Title: '.selectAll()'
+Description: 'A brief definition and examples of using selectAll()'
```
Comment (avdhoottt) [inline content/d3/concepts/selection/terms/selectAll/selectAll.md:3]: Add the actual description of the method in one line.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@craglin I'm sorry to hear that. Get well soon, I'm suggesting the final changes here.

**Comment by avdhoottt** [comment]
@craglin Hey, how is your health now? If you are going to take some more time or if you will not be able to complete this PR then please let me know. I'll talk to the internal team and decide then.

**Comment by craglin** [comment]
@avdhoottt Thanks for your understanding. Between my recovery, the holidays, and catching up on work I've been behind on my GitHub reviews. As far as I can see, you've already applied the changes your requested. Please let me know if there is anything else needed from me for final approval of this file.

**Comment by avdhoottt** [comment]
@craglin I just need one change in your entry. Can you please an example that is more informative than the current one? or the example which has some output to be shown.

File: content/d3/concepts/selection/terms/selectAll/selectAll.md [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+```
+
+- `selector`: Its a string version of an element, a saved node, or an array of nodes.
+
+## Example
+
+The following code selects all paragraph elements within the document using the `.selectAll()` method.
+
+```js
+const p = d3.selectAll("p")
```
Comment (avdhoottt) [inline content/d3/concepts/selection/terms/selectAll/selectAll.md:29]:
```suggestion
    // Select all elements with class 'example'
    const selectedElements = d3.selectAll('.example');

    // Modify the style of the selected elements
    selectedElements
      .style('color', 'blue')
      .style('font-size', '18px');

    // Add text to the selected elements
    selectedElements.text((d, i) => `Modified Element ${i + 1}`);

    // Log success message to console
    console.log('D3.js code executed successfully.');
```

**Comment by ishg-153** [comment]
@craglin @avdhoottt, how is the progress on this PR? Please let me know if I can help move things along.

**Review by ishg-153 — APPROVED** [review]
Ready to merge

**Review by avdhoottt — APPROVED** [review]
LGTM!

**Pull request merged**